### PR TITLE
[TRANSLATION] translate 6031.md and fix a tiny error

### PR DESCRIPTION
**Pull request by smxm**

### CODE REVIEW

File: docs/软件工程/6031.en.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- Class Hour: 100 hours
+
+The goal of this course is for you to learn how to write high quality code, and what is meant by high quality is to meet the following three targets:
+
+> Safe from bugs. Correctness (correct behavior right now) and defensiveness (correct behavior in the future) are required in any software we build.
+>
+> Easy to understand. The code has to communicate to future programmers who need to understand it and make changes in it (fixing bugs or adding new features). That future programmer might be you, months or years from now. You’ll be surprised how much you forget if you don’t write it down, and how much it helps your own future self to have a good design.
+>
+> Ready for change. Software always changes. Some designs make it easy to make changes; others require throwing away and rewriting a lot of code.
+
```
Comment (PKUFlyingPig) [inline docs/软件工程/6031.en.md:18]: 多了一个空行

File: docs/软件工程/6031.en.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+To accomplish this, the designers of this course have written a book that explains many of the core principles of software construction and the valuable lessons learned therefrom. The book covers such details as how to write annotations and function specifications, how to design abstract data structures, and many parallel programming aspects, all of which you will be able to explore and practice in carefully designed Java programming projects.
+
+The Spring 2016 semester course open-sources the code framework for all of its programming assignments, and the latest course materials can be found on its latest course website, see below for links.
+
+## Course Resources
+
+- Course Website: [2021spring](http://web.mit.edu/6.031/www/sp21/), [2016spring](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-005-software-construction-spring-2016/)
+- Recordings: None
```
Comment (PKUFlyingPig) [inline docs/软件工程/6031.en.md:27]: 有俩 recordings

File: docs/软件工程/6031.en.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+- Recordings: refer to the course website
+- Assignments：4 Problem Sets+ 1 Projects
```
Comment (PKUFlyingPig) [inline docs/软件工程/6031.en.md:29]: 4 Problem Sets + 1 Project

File: docs/软件工程/6031.en.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Personal Resources
+
+All the resources and assignments used by @PKUFlyingPig in this course are maintained in [PKUFlyingPig/MIT6.031 --- GitHub](https://github.com/PKUFlyingPig/MIT6.031-software-construction)
+
+@pengzhangzhi completed the assignment and recorded notes for this course, and the code is open source at [pengzhangzhi/self-taught-CS/Software Construction - Github](https://github.com/pengzhangzhi/self-taught-CS/tree/main/Software%20Construction)。
```
Comment (PKUFlyingPig) [inline docs/软件工程/6031.en.md:35]: @pengzhangzhi completed the assignments and took some notes, maintained at ...

File: docs/软件工程/6031.en.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+> Safe from bugs. Correctness (correct behavior right now) and defensiveness (correct behavior in the future) are required in any software we build.
+>
+> Easy to understand. The code has to communicate to future programmers who need to understand it and make changes in it (fixing bugs or adding new features). That future programmer might be you, months or years from now. You’ll be surprised how much you forget if you don’t write it down, and how much it helps your own future self to have a good design.
+>
+> Ready for change. Software always changes. Some designs make it easy to make changes; others require throwing away and rewriting a lot of code.
+
+
+To accomplish this, the designers of this course have written a book that explains many of the core principles of software construction and the valuable lessons learned therefrom. The book covers such details as how to write annotations and function specifications, how to design abstract data structures, and many parallel programming aspects, all of which you will be able to explore and practice in carefully designed Java programming projects.
```
Comment (PKUFlyingPig) [inline docs/软件工程/6031.en.md:20]: To achieve this, the instructors write a book explaining many of the core principles of software construction and valuable lessons learned from the past. The book covers many practical topics such as how to write comments and specifications, how to design abstract data structures, and many parallel programming caveats. You will explore all of these ideas in the programming assignments.

File: docs/软件工程/6031.en.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+The Spring 2016 semester course open-sources the code framework for all of its programming assignments, and the latest course materials can be found on its latest course website, see below for links.
```
Comment (PKUFlyingPig) [inline docs/软件工程/6031.en.md:22]: In the 2016 spring, the course open-sourced all of its programming assignments, and the textbook can be found on the latest website (see links below).

File: docs/软件工程/6031.en.md (hunk 1 above)
Comment (smxm) [reply docs/软件工程/6031.en.md:18]: 😳我看原文也有空行，删了空行那几句就并到一块了

Comment (PKUFlyingPig) [reply docs/软件工程/6031.en.md:18]: 好的～

File: docs/软件工程/6031.en.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+To achieve this, the instructors write a book explaining many of the core principles of software construction and valuable lessons learned from the past. The book covers many practical topics such as how to write comments and specifications, how to design abstract data structures, and many parallel programming caveats. You will explore all of these ideas in the programming assignments.
+
+In the 2016 spring, the course open-sourced all of its programming assignments, and the textbook can be found on the latest website (see links below).
+
+## Course Resources
+
+- Course Website: [2021spring](http://web.mit.edu/6.031/www/sp21/), [2016spring](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-005-software-construction-spring-2016/)
+- Recordings: None
+- Assignments：4 Problem Sets+ 1 Project
```
Comment (PKUFlyingPig) [inline docs/软件工程/6031.en.md:27]: 4 Problem Sets + 1 Project

Comment (PKUFlyingPig) [reply docs/软件工程/6031.en.md:27]: 空格

**Pull request merged**